src/librustdoc/clean/mod.rs

```diff
@@ -2271,12 +2260,20 @@ pub(crate) fn clean_middle_ty<'tcx>(
 
 fn clean_middle_opaque_bounds<'tcx>(
     cx: &mut DocContext<'tcx>,
-    bounds: Vec<ty::Clause<'tcx>>,
+    impl_trait_def_id: DefId,
+    args: ty::GenericArgsRef<'tcx>,
```

Comment:
Ended up inlining the `explicit_item_bounds` call into here, since I needed access to the def-id to call a query on it.